src/redis/driver_impl.go

```diff
@@ -77,9 +77,14 @@ func NewClientImpl(scope stats.Scope, useTls bool, auth, redisSocketType, redisT
 		}
 
 		if auth != "" {
-			logger.Warnf("enabling authentication to redis on %s", maskedUrl)
-
-			dialOpts = append(dialOpts, radix.DialAuthPass(auth))
+			user, pass, found := strings.Cut(auth, ":")
+			if found {
+				logger.Warnf("enabling authentication to redis on %s with user %s", maskedUrl, user)
+				dialOpts = append(dialOpts, radix.DialAuthUser(user, pass))
+			} else {
+				logger.Warnf("enabling authentication to redis on %s without user", maskedUrl)
+				dialOpts = append(dialOpts, radix.DialAuthPass(auth))
+			}
```

Comment:
Code looks sane but can you add tests for this?

Reply:
Thank you for the review. I have added some tests 😄